CameraXVideo/app/src/main/java/com/example/android/camerax/video/fragments/CameraFragment.kt

```diff
@@ -584,6 +600,45 @@ internal class QualityMap {
     }
 }
 
+/**
+ * a helper function to retrieve the aspect ratio from a QualitySelector enum.
+ */
+fun QualitySelector.getAspectRatio(quality:Int): Int {
+     return when {
+        arrayOf(QualitySelector.QUALITY_UHD,
+            QualitySelector.QUALITY_FHD,
+            QualitySelector.QUALITY_HD).contains(quality) -> AspectRatio.RATIO_16_9
+        quality ==  QualitySelector.QUALITY_SD -> AspectRatio.RATIO_4_3
+        else -> -1
```

Comment:
Maybe throw an UnsupportedOperationException here? Or handle all cases where -1 is returned.

Reply:
done with exception way. thanks!